promql/parser/parse.go

```diff
@@ -443,6 +467,147 @@ func (p *parser) newAggregateExpr(op Item, modifier, args Node) (ret *AggregateE
 	return ret
 }
 
+// Used when building the FloatHistogram from a map.
```

Comment:
Go doc comment style (which I think should be followed even if the function is not exporter): "newMap is used when building the FloatHistogram from a map."